upip/upip.py

```diff
@@ -157,6 +161,9 @@ def install_pkg(pkg_spec, install_path):
     f3 = tarfile.TarFile(fileobj=f2)
```

Comment:
To support debug logging, and potentially each of them may (should) have its own .close().